src/backend/utils/github-url.js

```diff
@@ -56,69 +56,70 @@ exports.getGithubUrlData = async incomingUrl => {
       : '';
+  try {
+    const fetchResult = await fetch(`${githubAPI}${subUrl}`);
+    const data = await fetchResult.json();
 
-  return fetch(`${githubAPI}${subUrl}`)
-    .then(res => res.json())
-    .then(data => {
-      let fetchedData;
+    let fetchedData;
+    /**
+     * Format the fetched data in a specifi way depending if the url
+     * is for a repo, user or for a {pull request, issue}
+     */
+    // User
+    if (ghUrl.repo === null) {
+      const { login: user, avatar_url: avatarURL, name, company, blog, email, bio } = data;
 
-      /**
-       * Format the fetched data in a specifi way depending if the url
-       * is for a repo, user or for a {pull request, issue}
-       */
-      // User
-      if (ghUrl.repo === null) {
-        const { login: user, avatar_url: avatarURL, name, company, blog, email, bio } = data;
+      fetchedData = {
+        user,
+        avatarURL,
+        name,
+        company,
+        blog,
+        email,
+        bio,
+      };
 
-        fetchedData = {
-          user,
-          avatarURL,
-          name,
-          company,
-          blog,
-          email,
-          bio,
-        };
+      // Repo
+    } else if (ghUrl.branch === 'master') {
+      const {
+        owner: { avatar_url: avatarURL },
+        description,
+        license,
+        open_issues: openIssues,
+        forks,
+        created_at: createdAt,
+        language,
+      } = data;
 
-        // Repo
-      } else if (ghUrl.branch === 'master') {
-        const {
-          owner: { avatar_url: avatarURL },
-          description,
-          license,
-          open_issues: openIssues,
-          forks,
-          created_at: createdAt,
-          language,
-        } = data;
+      fetchedData = {
+        avatarURL,
+        description,
+        license,
+        openIssues,
+        forks,
+        createdAt,
+        language,
+      };
 
-        fetchedData = {
-          avatarURL,
-          description,
-          license,
-          openIssues,
-          forks,
-          createdAt,
-          language,
-        };
+      // Issue or Pull Request
+    } else {
+      const {
+        user: { login, avatar_url: avatarURL },
+        body,
+        created_at: createdAt,
+      } = data;
 
-        // Issue or Pull Request
-      } else {
-        const {
-          user: { login, avatar_url: avatarURL },
-          body,
-          created_at: createdAt,
-        } = data;
-
-        fetchedData = {
-          login,
-          avatarURL,
-          body,
-          createdAt,
-          branch: ghUrl.branch,
-          repo: ghUrl.name,
-        };
-      }
-      return fetchedData;
-    });
+      fetchedData = {
+        login,
+        avatarURL,
+        body,
+        createdAt,
+        branch: ghUrl.branch,
+        repo: ghUrl.name,
+      };
+    }
+    return fetchedData;
+  } catch (error) {
+    throw new Error('Error fetching data');
```

Comment:
If all you're going to do in a `.catch` is `throw`, there's no value in doing it, and you can just leave it off.  However, it might be nice to log the error here, before you throw it.  Also, I'd consider just re-throwing the actual error vs. making your own, and losing the original error message and stack.

Reply:
What you do in  `src/backend/utils/inactive-blog-filter.js` below is what I mean.